deepchem/models/__init__.py

```diff
@@ -148,8 +154,195 @@ def predict(self, dataset, transformers=[]):
     # Remove padded examples.
     n_samples, n_tasks = len(dataset), len(self.tasks)
     y_pred = np.reshape(y_pred, (n_samples, n_tasks))
+    # Special case to handle singletasks.
+    if n_tasks == 1:
+      y_pred = np.reshape(y_pred, (n_samples,)) 
     return y_pred
 
+  def predict_grad(self, dataset, transformers=[]):
+    """
+    Uses self to calculate gradient on provided Dataset object.
+
+    TODO(rbharath): Should we assume each model has meaningful gradients to
```

Comment:
@rbharath This is added here because `HyperparamOpt` class returns the `Model` object as `best_model` rather than the `TensorFlowModel` object.  I think ideally this would be in the `TensorFlowAtomicRegressor` family of classes but due to how I was optimizing models and testing gradients I had to put it here.